content/client-lib-development-guide/features.textile

```diff
@@ -301,9 +301,9 @@ h3(#realtime-connection). Connection
 *** @(RTN15b1)@ @resume@ is the private connection key assigned to the connection when the first @CONNECTED@ @ProtocolMessage@ was received
 *** @(RTN15b2)@ @connection_serial@ is the most recent @ProtocolMessage#connectionSerial@ received from Ably or @Connection#serial@ which should be identical
 ** @(RTN15c)@ The system's response to a resume request will be one of the following:
-**** @(RTN15c1)@ @CONNECTED@ @ProtocolMessage@ with the same @connectionId@ as the current client, and no @error@. In this case, the server is indicating that the resume succeeded, all channels are still attached, and all backlog messages are available
-**** @(RTN15c2)@ @CONNECTED@ @ProtocolMessage@ with the same @connectionId@ as the current client, and an @error@. In this case, the server is indicating that the resume succeeded but with a non-fatal error, all channels are still attached, and some backlog messages may be unavailable. The @ErrorInfo@ received should be emitted on the client @Connection@ as well as all @ATTACHED@ @Channel@ objects.
-**** @(RTN15c3)@ @CONNECTED@ @ProtocolMessage@ with a new @connectionId@, and an error in @error@. In this case, a new connection has been established, the resume has failed, the channels are no longer attached, and the error indicates the resume problem which should be emitted. The client library should immediately detach all channels and set the @errorReason@ on each detached @Channel@
+**** @(RTN15c1)@ @CONNECTED@ @ProtocolMessage@ with the same @connectionId@ as the current client, and no @error@. In this case, the server is indicating that the resume succeeded, all channels are still attached, and all backlog messages are available. The client should not change the state of attached channels, and immediately process any queued messages for that channel
+**** @(RTN15c2)@ @CONNECTED@ @ProtocolMessage@ with the same @connectionId@ as the current client, and an @error@. In this case, the server is indicating that the resume succeeded but with a non-fatal error, all channels are still attached, and some backlog messages may be unavailable. The @ErrorInfo@ received should be emitted on the client @Connection@ as well as all @ATTACHED@ @Channel@ objects, and the @errorReason@ should be set. The client should not change the state of attached channels, and immediately process any queued messages for that channel
```

Comment:
I really don't like the idea that a) we don't know which channels are effected, b) we don't ever both notifying the user that an error has occurred, when discontinuity is indeed an error if you're expecting continuity on a channel. As a programmer, I think doing something like:

```javascript
channel = client.channels.get('x')
channel.on('attached', => { /* add channel to UI */ })
channel.on('detached', => { /* remove channel */ })
channel.on('failed', => { /* remove channel */ })
channel.on('error', => { /* notify user in channel messages are missing or get history */ })
```

is perfectly reasonable and expected.  However, expecting them to catch a connection error and KNOW that the connection error is related to the channel is just asking too much of a developer.  How will they know that an error message relates to the channel without looking up all our error codes, which is unrealistic for your average dev who just wants to catch an error and handle it.